src/plugins/console/public/application/hooks/use_data_init/use_data_init.test.ts

```diff
@@ -0,0 +1,201 @@
+  useEditorActionContext: jest.fn(),
+  useServicesContext: jest.fn(),
+}));
+
+jest.mock('./data_migration', () => ({
+  migrateToTextObjects: jest.fn(),
+}));
+
+import { renderHook } from '@testing-library/react-hooks';
+import { ContextValue, useEditorActionContext, useServicesContext } from '../../contexts';
+import { serviceContextMock } from '../../contexts/services_context.mock';
+import { migrateToTextObjects } from './data_migration';
+
+import { useDataInit } from './use_data_init';
+
+type UseStateMock = (...args: unknown[]) => [any, jest.Mock];
+const wait = (period: number) => new Promise((res) => setTimeout(res, period));
+
+describe('useDataInit', () => {
+  let mockContextValue: ContextValue;
+  let dispatch: (...args: unknown[]) => void;
+  let useStateMock: (...args: unknown[]) => [any, jest.Mock];
+  let setStateMock: jest.Mock;
+  const mockedObject = { mocked: true };
+
+  interface MockObjectData {
+    createdAt: number;
+    updatedAt: number;
+    text: string;
+  }
+
+  const callMockFuncWithArg = (arg: MockObjectData[] | Function | any) => ({
+    ...mockContextValue,
+    services: {
+      ...mockContextValue.services,
+      objectStorageClient: {
+        text: {
+          findAll: jest.fn(() => (typeof arg === 'function' ? arg() : arg)),
+          create: jest.fn((data: any) => mockedObject),
+        },
+      },
+    },
+  });
+
+  beforeAll(() => {
+    setStateMock = jest.fn();
+    useStateMock = (state: any) => [state, setStateMock];
+    mockContextValue = serviceContextMock.create();
+    dispatch = jest.fn();
+    // migrateToTextObjects = jest.fn();
+    (useEditorActionContext as jest.Mock).mockReturnValue(dispatch);
+    (useServicesContext as jest.Mock).mockReturnValue({
+      ...mockContextValue,
+      services: {
+        ...mockContextValue.services,
+        objectStorageClient: {
+          text: {
+            findAll: jest.fn(() => []),
+            create: jest.fn((data: any) => mockedObject),
+          },
+        },
+      },
+    });
+    // (migrateToTextObjects as jest.Mock).mockImplementation()
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    jest.restoreAllMocks();
+  });
+
+  afterAll(() => {
+    jest.resetAllMocks();
+  });
+
+  it('should calls dispatch with new objects if no texts are provided', async () => {
+    jest.spyOn(React, 'useState').mockImplementation(useStateMock);
+
+    renderHook(() => useDataInit(), {});
+
+    await wait(0);
+
+    expect(dispatch).toHaveBeenCalledWith({
+      type: 'setCurrentTextObject',
+      payload: mockedObject,
+    });
+  });
+
+  it('should calls dispatch with new objects if texts are provided', async () => {
+    jest.spyOn(React, 'useState').mockImplementation(useStateMock);
+
+    const mockObj: MockObjectData = {
+      createdAt: 1643277939899,
+      updatedAt: 1643277939900,
+      text: 'Mocked data',
+    };
+
+    (useServicesContext as jest.Mock).mockReturnValue(callMockFuncWithArg([mockObj]));
```

Comment:
Please, add more objects here in order to check if sorting is applied. And, please, add tests for this case.